### PR TITLE
[None][feat] Add draft-only top-k/top-p bypass for MTP

### PR DESCRIPTION
**Pull request by roborluo**
## Dev Engineer Review

- Current checkout changes five source files.
- Updates MTP-Eagle local full-vocabulary projection handling for ADP and LM-head tensor parallelism.
- Updates speculative metadata and configuration propagation for `draft_skip_top_k_top_p`.
- Raises an error when the required local full-vocabulary model hook is unavailable.
- No documentation, configuration, usage-manifest, or test changes are present in the current checkout.
- No current review severity findings were supplied.

## QA Engineer Review

No test changes.

## Per-File QA Perspective

- `tensorrt_llm/_torch/models/modeling_deepseekv3.py`: Changes MTP-head hidden-state preparation and local full-vocabulary projection. Verify token selection, output shape, and ADP behavior.
- `tensorrt_llm/_torch/models/modeling_qwen3_next.py`: Changes MTP-head projection and hidden-state handling. Verify final-token selection and logits across supported parallel configurations.
- `tensorrt_llm/_torch/speculative/eagle3.py`: Changes local MTP-head dispatch and unavailable-hook error handling. Verify supported heads and expected `RuntimeError` behavior.
- `tensorrt_llm/_torch/speculative/interface.py`: Changes draft sampling metadata and dispatch behavior. Verify greedy and stochastic paths and target sampling preservation.
- `tensorrt_llm/_torch/speculative/utils.py`: Changes speculative configuration validation and metadata propagation. Verify checkpoint-dependent mode resolution and option consistency.

## Description

Add default-off `MTPDecodingConfig.draft_skip_top_k_top_p` for PyTorch linear MTP. For non-greedy batches, this samples draft tokens using the request temperature without top-k/top-p filtering, with or without rejection sampling. All-greedy batches retain argmax, and target sampling settings remain unchanged.

Synchronize distributed draft dispatch and use local full-vocabulary logits for vanilla MTP under ADP + LM-head TP when either this flag or rejection sampling is enabled. This also corrects the logits layout for flag-off rejection sampling across greedy/stochastic transitions.

Include configuration, telemetry manifest entry, documentation, and regression tests. Existing rejection-sampling compatibility checks remain unchanged. Dynamic-tree MTP and AutoDeploy are not supported by this option.

## Test Coverage

- Configuration validation, metadata propagation, draft filtering, target sampling isolation, and proposal-probability storage.
- Vanilla MTP logits layout and greedy/stochastic transitions, including custom shared-head preprocessing.
- CUDA sampler correctness, mixed greedy requests, rejection-sampling distributions, and CUDA graph replay.
- Two-rank ADP + LM-head TP transitions with rejection sampling enabled and disabled.

Local validation: 148 source-isolated CPU tests passed, and pre-commit checks passed. Eleven GPU cases were skipped on this macOS host; native GPU/distributed validation remains pending. Full telemetry-manifest generation is also blocked locally by the missing CUDA dependency.

### Sampler-only benchmark

One Rubin SM 10.7 GPU, vocabulary 131072, temperature 0.8, top-p 0.95. Median GPU time for eight sequential draft steps, including proposal-probability storage. CUDA Graph replay, 12 alternating-order pairs, 100 repetitions per arm.

| Batch | Top-k | Skip OFF (µs) | Skip ON (µs) | Time reduction |
|---:|---:|---:|---:|---:|
| 1 | 0 (disabled) | 853.4 | 253.7 | 70.3% |
| 1 | 50 | 1052.5 | 253.6 | 75.9% |
| 8 | 0 (disabled) | 1021.7 | 354.4 | 65.3% |
| 8 | 50 | 1228.4 | 353.4 | 71.2% |

These timings exclude model forward, target verification, scheduling, and communication—not end-to-end speedups.

### Full-serving comparison

GLM-5.2, CTX4/GEN8, concurrency 1, MTP8 with rejection sampling. Jobs 581965 versus 583984; 240 seconds settling and ≥1500 seconds measurement.

## PR Checklist

- Documentation and regression tests included.
- No new dependencies.
- Add the `api-compatible` label for the new optional configuration field.
- Telemetry/privacy CODEOWNER approval is required.
- Native GPU CI validation is required before merge.

- [x] Reviewed the applicable PR checklist items and confirmed the required reviewers.

## GitHub Bot Help

To see available CI bot commands, comment `/bot help`.

### CODE REVIEW

**Review by Mgluhovskoi — APPROVED** [review]

File: tensorrt_llm/llmapi/llm_args.py [hunk 1]
```diff
@@ -2764,6 +2764,15 @@ def spec_dec_mode(self):
 
 class MTPDecodingConfig(DecodingBaseConfig):
     decoding_type: Literal["MTP"] = Field(default="MTP")
+    draft_skip_top_k_top_p: bool = Field(
```
Comment (BowenFu) [inline tensorrt_llm/llmapi/llm_args.py:2767]: Adding this public field to `MTPDecodingConfig` is what the `Check LLM API Compatibility Label` job keys on, and it's currently red with no labels set on the PR at all. Since the field defaults to `False` and leaves every existing path alone, would it be worth tagging this `api-compatible` to clear it?

Unrelated to the label: parametrizing the new tests over `skip=[False, True]` was a good call — I think that's what makes the `_compute_draft_logits` layout correction checkable against the old rejection-sampling behavior rather than just the new flag.

This one is required for the PR, since that check gates merge.

**Comment by roborluo** [comment]
There is one test 
> ADP + LM-head TP2 + mixed greedy/stochastic ranks

needs mult GPU here. So please add flag to make sure this function is correct. Thanks!

**Review by mikeiovine — CHANGES_REQUESTED** [review]
~This is already implemented by `AdvancedSamplingMode.no_topk_no_topp`~ Oh, I see, this is draft only? What's the reason for adding it?

**Comment by roborluo** [comment]
Thanks a lot!cloesed
